go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

```diff
@@ -662,9 +662,11 @@ func (zic *ZoneInterconnectHandler) addRemoteNodeStaticRoutes(node *corev1.Node,
 		}
 	}
 
-	if zic.IsSecondary() {
+	if zic.IsSecondary() && !(util.IsNetworkSegmentationSupportEnabled() && zic.IsPrimaryNetwork()) {
```

Comment:
note to self, we dont need to conditionalize this on topology being layer3, cause (zic *ZoneInterconnectHandler) AddRemoteZoneNode only gets called for default controller and layer 3 secondary net